src/crafting.cpp

```diff
@@ -1457,76 +1127,50 @@ bool player::disassemble( int dis_pos )
     return disassemble( i_at( dis_pos ), dis_pos, false );
 }
 
-bool player::disassemble( item &dis_item, int dis_pos,
-                          bool ground, bool msg_and_query )
+bool player::disassemble( item &obj, int pos, bool ground, bool interactive )
 {
-    const recipe *cur_recipe = get_disassemble_recipe( dis_item.typeId() );
+    const auto &r = recipe_dictionary::get_uncraft( obj.typeId() );
 
-    // No disassembly without proper light
-    // But book-ripping is OK
-    if( cur_recipe != nullptr &&
-        lighting_craft_speed_multiplier( *cur_recipe ) == 0.0f ) {
-        add_msg( m_info, _( "You can't see to craft!" ) );
-        activity.type = ACT_NULL;
+    // check sufficient light
+    if( lighting_craft_speed_multiplier( r ) == 0.0f ) {
```

Comment:
`==` float comparisons are considered bad practice, would be better to use `<=`.